lib/main.coffee

```diff
@@ -14,14 +33,36 @@ module.exports =
       (executablePath) =>
         @executablePath = executablePath
 
+    # Add a listener to reload this package if the custom grammar scopes setting
+    # changes. Grammar scopes can only be set at load time.
+    @subscriptions.add atom.config.observe 'linter-tidy.customGrammarScopes',
+      () ->
+        # Only reload the package if it is not already active.
+        # Otherwise, Atom could get stuck in an infinite loop.
+        if !atom.packages.isPackageActive('linter-tidy')
+          return
+
+        atom.packages.deactivatePackage('linter-tidy')
+        atom.packages.activatePackage('linter-tidy')
+
   deactivate: ->
     @subscriptions.dispose()
 
   provideLinter: ->
     helpers = require('atom-linter')
     regex = /line (\d+) column (\d+) - (Warning|Error): (.+)/g
+    grammarScopes = defaultGrammarScopes.slice()
```

Comment:
This whole block of code should just be removed, it will be handled with the changes to the `activate()` code.